test/conftest.py

```diff
@@ -60,3 +63,23 @@ def patch_mongomock(monkeypatch):
     monkeypatch.setattr(pymongo, 'MongoClient', MongoClient)
 
     monkeypatch.setitem(CONNECTIONS, 'pymongo', MongoClient)
+
+
+@pytest.fixture
```

Comment:
This means that we will be reconnecting to the db for every test. This should slow the tests quite a bit.

(Initial evidence seems to agree with this. The CI for this PR was on average 20s longer than previous tests.)

Reply:
I've found the difference to be somewhere around 10%. That's not, great, but should be alleviated when we do 
`mongomock`.